src/TypeSpec.Extension/Emitter.Csharp/src/emitter.ts

```diff
@@ -1,205 +1,135 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
-import { createSdkContext } from "@azure-tools/typespec-client-generator-core";
-import {
-    EmitContext,
-    Program,
-    logDiagnostics,
-    resolvePath
-} from "@typespec/compiler";
+import { EmitContext, Program, resolvePath } from "@typespec/compiler";
 
 import { execSync } from "child_process";
 import fs, { existsSync } from "fs";
-import { PreserveType, stringifyRefs } from "json-serialize-refs";
 import path from "node:path";
-import {
-    configurationFileName,
-    tspOutputFileName,
-    packageName
-} from "./constants.js";
-import { createModel } from "./lib/client-model-builder.js";
+import { configurationFileName, tspOutputFileName } from "./constants.js";
 import { LoggerLevel } from "./lib/log-level.js";
 import { Logger } from "./lib/logger.js";
 import {
     NetEmitterOptions,
     resolveOptions,
     resolveOutputFolder
 } from "./options.js";
-import { Configuration } from "./type/configuration.js";
+import { loadLibrary } from "./lib/load-module.js";
+import * as url from "url";
 
 export async function $onEmit(context: EmitContext<NetEmitterOptions>) {
     const program: Program = context.program;
     const options = resolveOptions(context);
-    const outputFolder = resolveOutputFolder(context);
-
     /* set the loglevel. */
     Logger.initialize(program, options.logLevel ?? LoggerLevel.INFO);
 
-    if (!program.compilerOptions.noEmit && !program.hasError()) {
-        // Write out the dotnet model to the output path
-        const sdkContext = createSdkContext(context, packageName);
-        const root = createModel(sdkContext);
+    /* load microsft csharp emitter. */
+    const basedir = url.fileURLToPath(new URL(".", import.meta.url));
+    Logger.getInstance().info(`basedir: ${basedir}.`);
+    const emitterNameOrPath = "@typespec/http-client-csharp";
```

Comment:
As @typespec/http-client-csharp is added into package.json,  can we call the exported $onEmit directly here instead of writing code to load the module/entrypoint?